core/src/main/java/org/opensearch/sql/calcite/utils/CalciteToolsHelper.java

```diff
@@ -113,15 +115,16 @@ public static RelBuilder create(
             new OpenSearchRelBuilder(config.getContext(), cluster, relOptSchema));
   }
 
-  public static Connection connect(FrameworkConfig config, JavaTypeFactory typeFactory) {
+  public static Connection connect(
+      FrameworkConfig config, JavaTypeFactory typeFactory, Integer querySizeLimit) {
     final Properties info = new Properties();
     if (config.getTypeSystem() != RelDataTypeSystem.DEFAULT) {
       info.setProperty(
           CalciteConnectionProperty.TYPE_SYSTEM.camelName(),
           config.getTypeSystem().getClass().getName());
     }
     try {
-      return new OpenSearchDriver().connect("jdbc:calcite:", info, null, typeFactory);
+      return new OpenSearchDriver(querySizeLimit).connect("jdbc:calcite:", info, null, typeFactory);
```

Comment:
Could we append Limit operator on each query instead of OpenSearchDriver constructor.

Reply:
Option 1 and 2 sounds both fine. Option 1 is more like a specific optimization for the setting query_size. I prefer to option 2.

Reply:
+1 on option2, 

> CONS: Will change the final plan if Limit operator cannot be push down

Could you elaborate on this? Are you suggesting that adding a LIMIT clause would change the result of the EXPLAIN plan? If that’s the concern, I’m okay with it. We explicitly enforce a querySizeLimit for every query anyway.


Latest impl is better, scanWithLimit is clean to m.